src/main/java/igrad/model/requirement/Credits.java

```diff
@@ -5,24 +5,26 @@
 
 /**
  * Represents a Requirement's required credit units in the course book.
- * Guarantees: immutable; is valid as declared in {@link #isValidCredits(String)}
+ * Guarantees: immutable, non-null and is valid as declared by {@link #isValidCredits(String)}
  */
 public class Credits {
 
-    public static final String MESSAGE_CONSTRAINTS = "Modular credits should contain one or more digits!";
+    public static final String MESSAGE_CONSTRAINTS =
+        "Credits needed to satisfy requirement should be more than 0.";
 
-    public static final String VALIDATION_REGEX = "\\d+";
+    public static final String VALIDATION_REGEX = "^[0-9]\\d*$"; // allows any numbers more than or equals zero
```

Comment:
the comments are abit misleading, at first i thought the regex allowed zero too

Reply:
oh wait actually it allowed 0 too, suggestion; `^[1-9]\\d*$`

Reply:
it allows zero.
i dont want to reject users who put 04 or 024 or something.

Reply:
then i guess you've to think of some regex to prevent '0'